pkg/ss/common/tps.go

```diff
@@ -15,3 +15,14 @@ type ClientMsg interface {
 	ReadOnly() bool
 	Key() core.KeyT
 }
+
+// todo ideally, we don't want to differentiate between replicated message and a client message.
+// but since the messages are already applied by the primary actor, we want to differentiate the two
+// so that the actor receiving this can treat it as a no-op
+type ReplicatedMsg struct {
```

Comment:
why not store an enum variable in ClientMsg to indicate if it is replicated or proxied.
The problem with inheritance is that it does not work well with serialization. if u have to send something on wire or disk u need explicit variables

Reply:
added a todo